mitodl/mit-open
# Browse by Topics section for the home page

### What are the relevant tickets?




[Homepage: "Browse by Topics" #4185](https://github.com/mitodl/hq/issues/4185)

### Description (What does it do?)


Adds the "Browse by Topics" section to the homepage:

[Desktop designs](https://www.figma.com/file/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?type=design&node-id=1009-18392&mode=design)
[Mobile designs](https://www.figma.com/file/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?type=design&node-id=1009-19640&mode=design&t=ANkF4Rqz61CYPap7-0)

### Screenshots (if appropriate):

<img width="1685" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/2d1a8fdd-6d1b-475c-a883-b73ff8ddb49a">
<img width="787" alt="image" src="https://github.com/mitodl/mit-open/assets/939376/9da422bf-a689-464a-b71a-fddac17804f3">


### How can this be tested?


Ensure topics are loaded if not already:
```bash
docker compose web python manage.py update_topics
```

- Load the homepage.

- Ensure the "Browse by Topics" section renders according to designs for desktop and for mobile breakpoint.

- Ensure the topic boxes link to the correct topic pages.

Notes:

- The "See all" button doesn't link anywhere (pending Topics page).

- We currently don't have any way to determine each topic's icon - they're currently using a random list. We'll need to either add this to the `GET /api/v1/topics/?is_toplevel=true` (for root topics), or perhaps hardcode if they're are known and unlikely to change often.
